### PR TITLE
BUILD-234: Add cross platform build support in Makefile

### PR DESCRIPTION
**Pull request by adambkaplan**
- Allow target `GOOS` and `GOARCH` to be configurable with
  `TARGET_GOOS` and `TARGET_GOARCH` variables
- Add race detection only if compiling for amd64.
- Remove ldflags that break compilation on s390x.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -72,7 +72,7 @@ verify: ## Run verifications. Example: make verify
 .PHONY: verify
 
 build: ## Build the executable. Example: make build
-	go build -a -mod=vendor -race -ldflags $(LDFLAGS) -o _output/csi-driver-shared-resource ./cmd
```
Comment (gabemontero) [inline Makefile:75]: could you add a `build-race` make target that still let's one do the race tests when they are on platforms that support it @adambkaplan

Comment (otaviof) [reply Makefile:75]: I just sent a [small PR to tackle `-race` flag](https://github.com/adambkaplan/openshift-csi-driver-shared-resource/pull/1) on different platform, so we can keep a single `build` target on the `Makefile`. Please consider.

Comment (gabemontero) [reply Makefile:75]: oooh even better :-) ...... thanks @otaviof 

and got to admit, I employed such platform checks quite a bit in my now distant past, programming in C and C++, and totally forgot about them; in my defense, @otaviof is much younger than me and undoubtedly has a better memory :-)

Comment (otaviof) [reply Makefile:75]: Thanks, Gabe! Granted some "creative stealing" to find a good example out there. :-)

**Comment by adambkaplan** [comment]
@gabemontero @otaviof updated Makefile such that we can compile on Power and Z.

I may want to follow up with a `verify` test which cross-compiles the CSI driver on amd64, arm, ppc64le, and s390x

File: Makefile (hunk 1 above)
Comment (adambkaplan) [reply Makefile:75]: done!

**Comment by gabemontero** [comment]
unrelated flake

/retest

/lgtm

**Pull request merged**